rust/private/rustc.bzl

```diff
@@ -320,17 +320,34 @@ def rustc_compile_action(
     # and use `$(pwd)` which resolves the `exec_root` at action execution time.
     package_dir = ctx.build_file_path[:ctx.build_file_path.rfind("/")]
     manifest_dir_env = "CARGO_MANIFEST_DIR=$(pwd)/{} ".format(package_dir)
-    command = '{}{}{} "$@" --remap-path-prefix="$(pwd)"=__bazel_redacted_pwd'.format(
-        manifest_dir_env,
-        out_dir_env,
-        toolchain.rustc.path,
+    maybe_rename = ""
+    if crate_info.type == "bin":
+        generated_file = crate_info.name
+        if toolchain.target_arch == "wasm32":
+            generated_file = generated_file + ".wasm"
+        src = "/".join([output_dir, generated_file])
+        dst = crate_info.output.path
+        if src != dst:
+            maybe_rename = " && /bin/mv {src} {dst}".format(src=src, dst=dst)
+    command = '{manifest_dir_env}{out_dir_env}{rustc} "$@" --remap-path-prefix="$(pwd)"=__bazel_redacted_pwd{maybe_rename}'.format(
```

Comment:
This line is long, maybe use a triple quoted string?

Reply:
Reverted away from the the fstring-like templates